### PR TITLE
DNS URI support

### PR DESCRIPTION
**Pull request by pspacek**

### CODE REVIEW

File: install/ui/src/freeipa/dns.js [hunk 1]
```diff
@@ -1226,7 +1226,21 @@ IPA.dns.get_record_metadata = function() {
             ],
             adder_attributes: [],
             columns: ['txt_part_data']
-        }
+        },
+        {
+            name: 'urirecord',
+            attributes: [
+                'uri_part_priority',
```
Comment (pvoborni) [inline install/ui/src/freeipa/dns.js]: Thinking about the 'doc' text from API,
```javascript
'uri_part_priority',
```
can be replaced with standard field definition, e.g., with:
```javascript
{
    name: 'uri_part_priority', 
    tooltip: '@mo-param:dnsrecord:uri_part_priority:doc'
}
```

**Comment by mbasti-rh** [comment]
jslint failed please fix it (you can leave pep8 as is, to be consistent with current code)

File: install/share/dns.ldif [hunk 2]
```diff
@@ -11,7 +11,7 @@ ipaDNSVersion: 2
...
+aci: (targetattr = "a6record || aaaarecord || afsdbrecord || aplrecord || arecord || certrecord || cn || cnamerecord || dhcidrecord || dlvrecord || dnamerecord || dnsclass || dnsttl || dsrecord || hinforecord || hiprecord || idnsallowdynupdate || idnsallowquery || idnsallowsyncptr || idnsallowtransfer || idnsforwarders || idnsforwardpolicy || idnsname || idnssecinlinesigning || idnssoaexpire || idnssoaminimum || idnssoamname || idnssoarefresh || idnssoaretry || idnssoarname || idnssoaserial || idnsupdatepolicy || idnszoneactive || ipseckeyrecord || keyrecord || kxrecord || locrecord || mdrecord || minforecord || mxrecord || naptrrecord || nsecrecord || nsec3paramrecord || nsrecord || nxtrecord || ptrrecord || rprecord || rrsigrecord || sigrecord || spfrecord || srvrecord || sshfprecord || tlsarecord || txtrecord || urirecord || unknownrecord ")(target = "ldap:///idnsname=*,cn=dns,$SUFFIX")(version 3.0;acl "Update DNS entries in a zone";allow (write) userattr = "parent[0,1].managedby#GROUPDN";)
```
Comment (mbasti-rh) [inline install/share/dns.ldif]: In the same way, you have to update 40-dns.update file, otherwise upgrade will not work (don't forgot to remove original ACI there as well)

File: ipaserver/plugins/dns.py [hunk 3]
```diff
@@ -28,6 +28,7 @@
 
 import dns.name
 import dns.exception
+import dns.rdata
```
Comment (mbasti-rh) [inline ipaserver/plugins/dns.py]: Why is this import needed?

**Comment by mbasti-rh** [comment]
NACK, please see inline comments

**Comment by mbasti-rh** [comment]
I was able to add an invalid URI record
```
[root@vm-058-017 ~]# ipa dnsrecord-add test.zone. --uri-rec='0 0 trolo"lo'
Record name: test2
  Record name: test2
  URI record: 0 0 "trolo"lo"

[root@vm-058-017 ~]# dig +short test2.test.zone. URI
[root@vm-058-017 ~]# 

journalctl output
failed to parse RR entry:  resource record DN 'idnsname=test2,idnsname=test.zone.,cn=dns,dc=blabla' data '0 0 "trolo"lo"': extra input text
```

**Comment by pspacek** [comment]
I was playing with an idea of automatic escaping but it cannot be done with current record format: There is no way to distinguish alredy escaped text from a text which needs escaping. This totally breaks web UI edit workflow because it reads text from LDAP, fills text field with it and then double-escapes it during save.

For this reason I've given up attempts to automatically escape things, which is no different from e.g. TXT records. Call it consistency if you want ;-)

**Comment by mbasti-rh** [comment]
I'm so sorry, I didn't noticed earlier, but you forgot to bump API in VERSION

Otherwise LGTM and works for me

**Comment by pspacek** [comment]
I thought that messing with API numbers in VERSION is not be necessary anymore after we introduced thin client. @jcholast ?

**Comment by mbasti-rh** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/f363dfbeed7aeba51d694a60b29389d94c7bda44
https://fedorahosted.org/freeipa/changeset/bf96b80200c3e8d1795a913db4e0222ea558e609
https://fedorahosted.org/freeipa/changeset/51af7a15981eca753dfbda133cc557d0512f6e95
